libs/sdk-core/src/breez_services.rs

```diff
@@ -910,8 +910,21 @@ impl BreezServices {
             closed_channel_payments.push(closed_channel_tx);
         }
 
+        let pending_payments: Vec<Payment> = new_data
+            .payments
+            .clone()
+            .into_iter()
+            .filter(|c| c.status == PaymentStatus::Pending)
+            .collect();
+        let mut payments: Vec<Payment> = vec![];
+        if !pending_payments.is_empty() {
+            info!("pending paymetns are not empty {:?}", pending_payments);
+            payments.extend(self.node_api.htlc_expiry(pending_payments).await?);
+        }
+        info!("payment list consist of {:?}", payments);
```

Comment:
I think all this needs to be inside node_api. We don't need another method for `htlc_expiry` and we rather ensure the `pull_changed` method returns the payments with the right expiry populated.

Reply:
Moved this logic.